mdgen/src/main/java/uk/gov/ida/mdgen/MetadataGenerator.java

```diff
@@ -141,32 +125,19 @@ public Void call() throws Exception {
         return null;
     }
 
-    private BasicX509Credential getSigningCredentialFromCloudHSM(X509Certificate cert) throws Exception {
+    private BasicX509Credential getSigningCredentialFromCloudHSM(X509Certificate cert, String label) throws Exception {
+        KeyStore cloudHsmStore = KeyStore.getInstance("Cavium");
+        cloudHsmStore.load(null, null);
+        return new BasicX509Credential(cert, (PrivateKey) cloudHsmStore.getKey(label, null));
+    }
+
+    private void setSecurityProvider() throws InstantiationException, IllegalAccessException, java.lang.reflect.InvocationTargetException, NoSuchMethodException, ClassNotFoundException {
```

Comment:
`java.lang.reflect.InvocationTargetException` can be shortened to `InvocationTargetException`